File: Luogu/CodeForces Practice/cf538B/cf538B.cpp
#include<bits/stdc++.h>
using namespace std;
int ans[10];
int main(){
    int n,maxn=0;
    cin>>n;
    while(n){
        ans[++ans[0]]=n%10;
        maxn=max(maxn,ans[ans[0]]);
        n/=10;
    }
    cout<<maxn<<endl;
    for(int i=1;i<=maxn;i++){
        bool flag=false;
        for(int j=ans[0];j>0;j--){
            if(ans[j]>0){
                ans[j]--;
                cout<<1;
                flag=true;
            }else{
                if(flag)cout<<0;
            }
        }
        cout<<" ";
    }
    cout<<endl;
    return 0;
}

File: CodeForces Practice/CF538B/CF538B.java
//import java.util.Arrays;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		int[] ans = new int[15];
		int maxn = 0;
		ans[0] = 0;
		while (n > 0) {
			ans[++ans[0]] = n % 10;
			maxn = max(maxn, ans[ans[0]]);
			n /= 10;
		}
		System.out.printf("%d\n", maxn);
		for (int i = 1; i <= maxn; i++) {
			boolean flag = false;
			for (int j = ans[0]; j >= 1; j--) {
				if (ans[j] > 0) {
					ans[j]--;
					System.out.printf("%d", 1);
					flag = true;
				} else {
					if (flag) System.out.printf("%d", 0);
				}
			}
			System.out.print(" ");
		}
		System.out.println("");
	}

	private static int max(int a, int b) {
		if (a >= b) return a; else return b;
	}
}
